src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -335,29 +335,32 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
         {
             assert(hasImmediateOperand);
 
-            if (isRMW)
-            {
-                GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
-
-                HWIntrinsicImmOpHelper helper(this, intrin.op3, node);
+            auto emitShift = [&](GenTree* op, regNumber reg) {
+                HWIntrinsicImmOpHelper helper(this, op, node);
 
                 for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
                 {
                     const int shiftAmount = helper.ImmValue();
 
-                    GetEmitter()->emitIns_R_R_I(ins, emitSize, targetReg, op2Reg, shiftAmount, opt);
+                    if (shiftAmount == 0)
+                    {
+                        GetEmitter()->emitIns_R_R_R(INS_mov, emitTypeSize(node), targetReg, reg, reg);
```

Comment:
We really want to use `emitIns_Mov` but there is an issue with the ARM64 impl that will elide the `mov` even if `canSkip` is `false`.

Reply:
Why is eliding a mov an issue here?

Reply:
It breaks the jump table logic which expects a fixed number of bytes per switch case (on Arm64 that's 2 instructions per case, so 8-bytes)